### ISSUE TITLE
2 vulnerabilities in npm packages in newly generated app

### ISSUE THREAD

**Issue by prog1dev** [expo/expo#2472]
After generating new app and `npm i` I got this message in console:
```
found 2 vulnerabilities (1 low, 1 high) 
  run `npm audit fix` to fix them, or `npm audit` for details
```

`npm audit fix` does not help:
```
fixed 0 of 2 vulnerabilities in 13948 scanned packages
  2 vulnerabilities required manual review and could not be updated
```

Here is a full `npm audit` output:
```
                                                                                
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Denial of Service                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ ws                                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >= 1.1.5 <2.0.0 || >=3.3.1                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ b53a5eeaadeb664472d88424cf00a2f9d89aa62313f4fda3ebc7c35c2fb… │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ b53a5eeaadeb664472d88424cf00a2f9d89aa62313f4fda3ebc7c35c2fb… │
│               │ > react-devtools-core > ws                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/550                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.5                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ b53a5eeaadeb664472d88424cf00a2f9d89aa62313f4fda3ebc7c35c2fb… │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ b53a5eeaadeb664472d88424cf00a2f9d89aa62313f4fda3ebc7c35c2fb… │
│               │ > plist > xmlbuilder > lodash                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/577                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 2 vulnerabilities (1 low, 1 high) in 13948 scanned packages
  2 vulnerabilities require manual review. See the full report for details.

```

## Steps to reproduce

```
expo init AwesomeProject && cd AwesomeProject && npm i
```

## Environment

```
Environment:
  OS: macOS High Sierra 10.13.4
  Node: 8.9.1
  Yarn: 1.7.0
  npm: 6.4.1
  Watchman: Not Found
  Xcode: Xcode 9.3 Build version 9E145
  Android Studio: Not Found

Packages: (wanted => installed)
  expo: ^30.0.1 => 30.0.1
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz => 0.55.4
```

**Comment by brentvatne** [comment]
ws is a downstream dependency of react-devtools and only used in development if you use react-devtools. we'll update that dependency along with react-native as it controls the version.

re: lodash, this is not really anything the user needs to be concerned about as it's depended on by expo devtools transitively through the plist package. cc @fson - it might be worth updating just to get rid of the warning

**Comment by brentvatne** [comment]
> Is there anything we can do about this in the short-term while we await the next Expo release?

you could use yarn resolutions to override but these are not actual security concerns, they do not impact your application or your at all, so you don't need to worry about them. you can't use the latest version of react-native with expo yet, sdk31 will have that.

> What's the best way to ignore/uninstall Expo modules I don't need to use?

currently there's no built-in way to do this. you could fork the expo sdk javascript package if you like and remove the dependencies you don't need though.

**Comment by calumjames** [comment]
Thanks a lot for the helpful responses, @brentvatne!

It's more to do with me being pedantic about a clean `npm` output rather than me being concerned about security issues in this case 😄 Maybe I can get over it for now though until the next Expo SDK is released.

**Comment by incarnateTheGreat** [comment]
Hey there. Also experiencing the same issue: I did an `npm i` yesterday while adding a new dependency and now I'm experiencing this issue.

[This post](https://github.com/react-community/create-react-native-app/issues/775#issuecomment-423943293) talks about updating the dependencies in **react-devtools-core**, but I can't seem to do it.

**Comment by sjchmiela** [comment]
A newly created project shows 0 vulnerabilities now, yay! 🎉 

```
                                                                                
                       === npm audit security report ===                        
                                                                                
found 0 vulnerabilities
 in 20901 scanned packages
```

**Comment by surajchaudhari** [comment]
=== npm audit security report ===


                                 Manual Review
             Some vulnerabilities require your attention to resolve

          Visit https://go.npm.me/audit-guide for additional guidance


  Low             Regular Expression Denial of Service

  Package         debug

  Patched in      >= 2.6.9 < 3.0.0 || >= 3.1.0

  Dependency of   botpress

  Path            botpress > socket.io > debug

  More info       https://nodesecurity.io/advisories/534


  Low             Regular Expression Denial of Service

  Package         debug

  Patched in      >= 2.6.9 < 3.0.0 || >= 3.1.0

  Dependency of   botpress

  Path            botpress > socket.io > engine.io > debug

  More info       https://nodesecurity.io/advisories/534


  Low             Regular Expression Denial of Service

  Package         debug

  Patched in      >= 2.6.9 < 3.0.0 || >= 3.1.0

  Dependency of   botpress

  Path            botpress > socket.io > socket.io-adapter > debug

  More info       https://nodesecurity.io/advisories/534


  Low             Regular Expression Denial of Service

  Package         debug

  Patched in      >= 2.6.9 < 3.0.0 || >= 3.1.0

  Dependency of   botpress

  Path            botpress > socket.io > socket.io-client > debug

  More info       https://nodesecurity.io/advisories/534


  Low             Regular Expression Denial of Service

  Package         debug

  Patched in      >= 2.6.9 < 3.0.0 || >= 3.1.0

  Dependency of   botpress

  Path            botpress > socket.io > socket.io-client > engine.io-client >
                  debug

  More info       https://nodesecurity.io/advisories/534


  Low             Regular Expression Denial of Service

  Package         debug

  Patched in      >= 2.6.9 < 3.0.0 || >= 3.1.0

  Dependency of   botpress

  Path            botpress > socket.io > socket.io-adapter > socket.io-parser
                  > debug

  More info       https://nodesecurity.io/advisories/534


  Low             Regular Expression Denial of Service

  Package         debug

  Patched in      >= 2.6.9 < 3.0.0 || >= 3.1.0

  Dependency of   botpress

  Path            botpress > socket.io > socket.io-client > socket.io-parser >
                  debug

  More info       https://nodesecurity.io/advisories/534


  Low             Regular Expression Denial of Service

  Package         debug

  Patched in      >= 2.6.9 < 3.0.0 || >= 3.1.0

  Dependency of   botpress

  Path            botpress > socket.io > socket.io-parser > debug

  More info       https://nodesecurity.io/advisories/534


  Moderate        Prototype pollution

  Package         hoek

  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3

  Dependency of   botpress

  Path            botpress > jsonwebtoken > joi > hoek

  More info       https://nodesecurity.io/advisories/566


  Moderate        Prototype pollution

  Package         hoek

  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3

  Dependency of   botpress

  Path            botpress > jsonwebtoken > joi > topo > hoek

  More info       https://nodesecurity.io/advisories/566


  High            Regular Expression Denial of Service

  Package         parsejson

  Patched in      No patch available

  Dependency of   botpress

  Path            botpress > socket.io > socket.io-client > engine.io-client >
                  parsejson

  More info       https://nodesecurity.io/advisories/528

found 11 vulnerabilities (8 low, 2 moderate, 1 high) in 5223 scanned packages
  11 vulnerabilities require manual review. See the full report for details.

D:\Botpress Data>
